be/src/runtime/fragment_mgr.cpp

```diff
@@ -571,14 +579,24 @@ void FragmentMgr::set_pipe(const TUniqueId& fragment_instance_id,
     }
 }
 
-std::shared_ptr<io::StreamLoadPipe> FragmentMgr::get_pipe(const TUniqueId& fragment_instance_id) {
+std::shared_ptr<io::StreamLoadPipe> FragmentMgr::get_pipe(const TUniqueId& fragment_instance_id,
+                                                          bool enable_pipeline_engine) {
     {
         std::lock_guard<std::mutex> lock(_lock);
-        auto iter = _fragment_map.find(fragment_instance_id);
-        if (iter != _fragment_map.end()) {
-            return _fragment_map[fragment_instance_id]->get_pipe();
+        if (enable_pipeline_engine) {
```

Comment:
maybe should do double search and remove useless `enable_pipeline_engine` in function param and thrift struct